log/logger_test.go

```diff
@@ -14,6 +15,29 @@ const (
 	logName = "test"
 )
 
+func TestNewLoggerError(t *testing.T) {
+	// we expect an error from NewLoggerE in the event that we provide an
+	// unwriteable directory
+
+	// this test needs a guaranteed empty directory, so we create a temporary one
+	// and ensure that it gets destroyed afterward.
+	targetDir := os.TempDir()
+	t.Cleanup(func() {
+		// This removal could produce an error, but since it's a temporary
+		// directory anyway, this is a best-effort cleanup
+		os.Remove(targetDir)
+	})
```

Comment:
my reading of [os.TempDir()](https://pkg.go.dev/os#TempDir) is that it will return `/tmp`, and i don't think deleting _that_ is what you want to do here

Reply:
Ah, yep--good catch. `os.MkdirTemp` is what I wanted. Yay code reviews :)